Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp

```diff
@@ -125,20 +125,22 @@ void MediaSourcePrivateRemote::bufferedChanged(const PlatformTimeRanges& buffere
     m_gpuProcessConnection->connection().send(Messages::RemoteMediaSourceProxy::BufferedChanged(buffered), m_identifier);
 }
 
-void MediaSourcePrivateRemote::markEndOfStream(EndOfStreamStatus)
+void MediaSourcePrivateRemote::markEndOfStream(EndOfStreamStatus status)
 {
-    notImplemented();
+    m_ended = true;
+    m_gpuProcessConnection->connection().send(Messages::RemoteMediaSourceProxy::MarkEndOfStream(status), m_identifier);
 }
 
 void MediaSourcePrivateRemote::unmarkEndOfStream()
 {
-    notImplemented();
+    // FIXME(125159): implement unmarkEndOfStream()
+    m_ended = false;
+    m_gpuProcessConnection->connection().send(Messages::RemoteMediaSourceProxy::UnmarkEndOfStream(), m_identifier);
 }
 
 bool MediaSourcePrivateRemote::isEnded() const
 {
-    notImplemented();
-    return false;
+    return m_ended;
```

Comment:
If all implementations of markEndOfStream/unmarkEndOfStream/isEnded are doing the same with regards to m_ended, maybe this should be moved to MediaSourcePrivate and we would have markEndOfStreamInternal/unmarkEndOfStreamInternal methods that we would specialize

Reply:
good point. I'll check other implementations and do that if possible.

Reply:
MediaSourcePrivate is currently a pure virtual class, I don't really want to change this at this stage.